medusa/search/core.py

```diff
@@ -476,87 +475,104 @@ def wanted_episodes(series_obj, from_date):
+        )
         cur_provider.cache.update_cache()
 
+    single_results = {}
+    multi_results = []
     for cur_provider in providers:
-        threading.currentThread().name = u'{thread} :: [{provider}]'.format(thread=original_thread_name,
-                                                                            provider=cur_provider.name)
+        threading.currentThread().name = u'{thread} :: [{provider}]'.format(
+            thread=original_thread_name, provider=cur_provider.name
+        )
         try:
-            cur_found_results = cur_provider.search_rss(episodes)
+            found_results = cur_provider.cache.find_needed_episodes(episodes)
         except AuthException as error:
             log.error(u'Authentication error: {0}', ex(error))
             continue
 
         # pick a single result for each episode, respecting existing results
-        for cur_ep in cur_found_results:
-            if not cur_ep.series or cur_ep.series.paused:
-                log.debug(u'Skipping {0} because the show is paused ', cur_ep.pretty_name())
+        for episode_no, results in iteritems(found_results):
+            if results[0].series.paused:
+                log.debug(u'Skipping {0} because the show is paused.', results[0].series.name)
                 continue
 
             # if all results were rejected move on to the next episode
-            wanted_results = filter_results(cur_found_results[cur_ep])
+            wanted_results = filter_results(results)
             if not wanted_results:
-                log.debug(u'All found results for {0} were rejected.', cur_ep.pretty_name())
+                log.debug(u'All found results for {0} were rejected.', results[0].series.name)
                 continue
 
             best_result = pick_result(wanted_results)
-            # if it's already in the list (from another provider) and the newly found quality is no better then skip it
-            if cur_ep in found_results and best_result.quality <= found_results[cur_ep].quality:
-                continue
-
             # Skip the result if search delay is enabled for the provider.
             if delay_search(best_result):
                 continue
 
-            found_results[cur_ep] = best_result
+            if episode_no in (SEASON_RESULT, MULTI_EP_RESULT):
+                multi_results.append(best_result)
+            else:
+                # if it's already in the list (from another provider) and
+                # the newly found quality is no better then skip it
+                if episode_no in single_results:
+                    allowed_qualities, preferred_qualities = results[0].series.current_qualities
+                    if not Quality.is_higher_quality(single_results[episode_no].quality,
+                                                     best_result.quality, allowed_qualities,
+                                                     preferred_qualities):
+                        continue
```

Comment:
Log a debug message when a result is skipped.

Reply:
I wouldn't add this. It's going to be very spammy.